crashreports/internal/win32/CrashReportContext.cpp

```diff
@@ -0,0 +1,115 @@
+#include <locale>
+#include <codecvt>
+#include <sstream>
+#include "client/windows/handler/exception_handler.h"
+#include "CrashReportContext.h"
```

Comment:
When you write a .cpp / .h file pair, it is good to make the #include of the .h file the first of all #include directives.

That compels you to put all the necessary nested #include directives or forward declarations in that header, so that other source code files might also #include the header as a self-contained thing.

I noticed that one version of CrashReportContext.h has <map> but the other doesn't, but appears to need it because you include <map> before it in the .cpp file.

Reply:
I agree with @Paul-Licameli and we should dividing headers, that needed for `.h` file should included only into `.h` file and that needed for `.cpp` should be included only in `.cpp` after `#include <target>.h`.

Also @Paul-Licameli, @crsib, I think it will be great if we can using `path/to/file` conception in `#include` directive. That means when we have file tree (example): `src/base/base.h`, `src/modules/module_a.h` and `src/app.h`, all path in `#include` will be same for each.
For example:
We want to include `src/base/base.h` in `src/modules/module_a` and `src/app.h`.
Therefore, `#include` directive will be same for both as:
```
#include "base/base.h" // Good
#include "src/base/base.h" // Good

// No Good
#include "../base/base.h" // for src/modules/module_a
#include "base/base.h" // for src/app
```

Reply:
@gera-gas I'm not sure I understand you.

But if you mean paths with ../ are bad -- though now numerous under our src/ -- then I agree with that, and my big branch misc-modules eliminates many of those.

It simplifies almost all inclusions to just a file name, with no directory names, by supplying appropriate -I options on the compiler command line.